sky/provision/aws/instance.py

```diff
@@ -1,17 +1,30 @@
 """AWS instance provisioning."""
+import re
+import time
 from typing import Any, Dict, List, Optional
 
 from botocore import config
+from botocore import exceptions
 
+from sky import sky_logging
 from sky import status_lib
 from sky.adaptors import aws
 from sky.utils import common_utils
 
+logger = sky_logging.init_logger(__name__)
+
 BOTO_MAX_RETRIES = 12
 # Tag uniquely identifying all nodes of a cluster
 TAG_RAY_CLUSTER_NAME = 'ray-cluster-name'
 TAG_RAY_NODE_KIND = 'ray-node-type'
 
+INITIAL_BACKOFF_SECONDS = 7
+MAX_BACKOFF_FACTOR = 10
```

Comment:
Where does these numbers come from?

Reply:
It is adopted from one of my previous PR #2265 🤔 We need to determine the best numbers here

Basically, I bisect the `INITIAL_BACKOFF_SECONDS` and keep other numbers the same, and ensure it could correctly cleanup even if we don't do the `instance.wait_until_terminated()`.

Reply:
Make sense! Adopted. Thanks 😉